artifact/scripts/executor.hpp

```diff
@@ -0,0 +1,138 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_ARTIFACT_V3_SCRIPT_EXECUTOR_HPP
+#define MENDER_ARTIFACT_V3_SCRIPT_EXECUTOR_HPP
+
+#include <memory>
+#include <string>
+
+#include <common/common.hpp>
+#include <common/conf.hpp>
+#include <common/events.hpp>
+#include <common/expected.hpp>
+#include <common/log.hpp>
+#include <common/optional.hpp>
+#include <common/processes.hpp>
+#include <artifact/scripts/error.hpp>
+
+namespace mender {
+namespace artifact {
+namespace scripts {
+namespace executor {
+
+using namespace std;
+
+namespace events = mender::common::events;
+namespace expected = mender::common::expected;
+namespace log = mender::common::log;
+namespace optional = mender::common::optional;
+namespace processes = mender::common::processes;
+
+using Error = mender::common::error::Error;
+
+enum class State {
+	Idle,
+	Sync,
+	Download,
+	ArtifactInstall,
+	ArtifactReboot,
+	ArtifactCommit,
+	ArtifactRollback,
+	ArtifactRollbackReboot,
+	ArtifactFailure,
+};
+
+enum class Action {
+	Enter,
+	Leave,
+	Error,
+};
+
+class OutputHandler {
```

Comment:
Is this thing exactly the same as the original? Could we centralize it?

Reply:
Yeh, we probably should. I'll add it to `common` somewhere